gazelle ignores default goexperiments causing builds to fail

MRE: 

[gazelle-default-build-constarints-mre.zip](https://github.com/user-attachments/files/23463420/gazelle-default-build-constarints-mre.zip)

In this MRE a package builds selectively based on `goexperiments.swissmap` which as of go 1.24 is now [on by default](https://go.dev/doc/go1.24#runtime). This causes the build to fail as

1. gazelle looks for which files to include in `srcs`, determines no experiments are on (im assuming, perhaps it just hasnt been updated about swissmap), and only includes `fortune_noexp.go`
2. the go toolchain, building with the default swissmap experiment as on, does not compile `fortune_noexp.go` and does not find a file defining `var fortunes`, so it fails to compile
```
Computing main repo mapping: 
Loading: 
Loading: 0 packages loaded
Analyzing: target //fortune:fortune (0 packages loaded, 0 targets configured)
Analyzing: target //fortune:fortune (0 packages loaded, 0 targets configured)

INFO: Analyzed target //fortune:fortune (0 packages loaded, 0 targets configured).
ERROR: /home/azransayed/gazelle-build-constraints-mre/fortune/BUILD.bazel:3:11: GoCompilePkg fortune/fortune.a failed: (Exit 1): builder failed: error executing GoCompilePkg command (from target //fortune:fortune) bazel-out/k8-opt-exec-ST-d57f47055a04/bin/external/rules_go++go_sdk+main___download_0/builder_reset/builder compilepkg -sdk external/rules_go++go_sdk+main___download_0 -goroot ... (remaining 27 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
fortune/fortune.go:6:9: undefined: fortunes
compilepkg: error running subcommand external/rules_go++go_sdk+main___download_0/pkg/tool/linux_amd64/compile: exit status 2
Target //fortune:fortune failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 0.153s, Critical Path: 0.02s
INFO: 2 processes: 2 internal.
ERROR: Build did NOT complete successfully
```

I dont know if this behavior is desired, but bazel-contrib/rules_go#409 makes me think that this is a bug

--- a/language/go/build_constraints.go
+++ b/language/go/build_constraints.go
@@ -274,13 +274,13 @@ func matchAuto(tokens []string) (*buildTags, error) {
 	return newBuildTags(x), nil
 }
 
-// isDefaultIgnoredTag returns whether the tag is "cgo", "purego", "race", "msan"  or is a release tag.
+// isDefaultIgnoredTag returns whether the tag is "cgo", "purego", "race", "msan" , is a goexperiment, or is a release tag.
 // Release tags match the pattern "go[0-9]\.[0-9]+".
 // Gazelle won't consider whether an ignored tag is satisfied when evaluating
 // build constraints for a file and will instead defer to the compiler at compile
 // time.
 func isDefaultIgnoredTag(tag string) bool {
-	if tag == "cgo" || tag == "purego" || tag == "race" || tag == "msan" {
+	if tag == "cgo" || tag == "purego" || tag == "race" || tag == "msan" || strings.HasPrefix(tag, "goexperiments.") {
 		return true
 	}
 	if len(tag) < 5 || !strings.HasPrefix(tag, "go") {

--- a/language/go/fileinfo_go_test.go
+++ b/language/go/fileinfo_go_test.go
@@ -493,6 +493,42 @@ func TestShellSafety(t *testing.T) {
 	}
 }
 
+func TestGoExperimentsTagsIgnored(t *testing.T) {
+	dir, err := os.MkdirTemp(os.Getenv("TEST_TEMPDIR"), "TestGoExperimentsTagsIgnored")
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Cleanup(func() {
+		os.RemoveAll(dir)
+	})
+
+	for _, tc := range []struct {
+		desc, content string
+	}{
+		{
+			desc:    "goexperiments tag satisfied",
+			content: "//go:build goexperiments.range\n\npackage foo",
+		},
+		{
+			desc:    "negated goexperiments tag satisfied",
+			content: "//go:build !goexperiments.range\n\npackage foo",
+		},
+	} {
+		t.Run(tc.desc, func(t *testing.T) {
+			path := filepath.Join(dir, tc.desc+".go")
+			if err := os.WriteFile(path, []byte(tc.content), 0o600); err != nil {
+				t.Fatal(err)
+			}
+
+			c, _, _ := testConfig(t)
+			fi := goFileInfo(path, "")
+			if !checkConstraints(c, "", "", fi.goos, fi.goarch, fi.tags, nil) {
+				t.Fatalf("constraints should be satisfied for %s", tc.desc)
+			}
+		})
+	}
+}
+
 func mustParseBuildTag(t *testing.T, in string) constraint.Expr {
 	x, err := constraint.Parse("//go:build " + in)
 	if err != nil {

